--- 3rdparty/libpng/CMakeLists.txt	2021-04-02 19:23:54.000000000 +0800
+++ 3rdparty/libpng/CMakeLists.txt.new	2022-09-18 23:06:12.000000000 +0800
@@ -3,6 +3,10 @@
 #
 # ----------------------------------------------------------------------------
 
+if ( "${CMAKE_OSX_ARCHITECTURES}" MATCHES "arm64")
+  set(ENABLE_NEON ON)
+endif()
+
 if(ENABLE_NEON)
   project(${PNG_LIBRARY} C ASM)
 else()
@@ -28,7 +32,7 @@
 file(GLOB lib_hdrs *.h)
 
 
-if(ARM OR AARCH64)
+if(ARM OR AARCH64 OR ENABLE_NEON)
   if(ENABLE_NEON)
     if(NOT AARCH64)
       list(APPEND lib_srcs arm/filter_neon.S)
